src/app/features/observations/cells/VerificationCell.tsx

```diff
@@ -9,38 +9,77 @@ import {
 } from "@material-ui/icons";
 import React from "react";
 import { connect, DispatchProp } from "react-redux";
-import { TmpObservation } from "../../../../store/reducers/observationListReducer";
+import { setObservationVerificationStatus } from "../../../../store/actions/observationListActions";
+import { ObservationData, VerificationStatus } from "../models";
 
 export const VerificationCell = connect()(
   ({
     dispatch,
     observation
-  }: DispatchProp & { observation: TmpObservation }) => (
-    // TODO update api and wire up
+  }: DispatchProp & { observation: ObservationData }) => (
     <>
-      {observation.verified ? (
-        <IconButton className="p-0 mr-2" disableRipple>
-          <RadioButtonUnchecked />
-        </IconButton>
-      ) : (
+      {observation.verificationStatus === "Pending" ? (
         <IconButton className="p-0 mr-2 text-primary" disableRipple>
           <RadioButtonChecked />
         </IconButton>
+      ) : (
+        <IconButton
+          className="p-0 mr-2"
+          disableRipple
+          onClick={() =>
+            dispatch(
+              setObservationVerificationStatus.request({
+                id: observation.id,
```

Comment:
We could also have destructuring for observation since we repeatedly get the same keys from this object.
`const { id, verificationStatus } = observation;`
Same could be done for VerificationStatus.
`cosnt { Pending, Rejected, Verified } = VerificationStatus;`